src/convertToObject.js

```diff
@@ -1,18 +1,23 @@
 'use strict';
 
-/**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  const sortedSourceArray = sourceString
+    .replace(/\s+/g, ' ')
+    .trim()
+    .split(';')
+    .map(splitProperty => splitProperty.split(':'))
+    .filter(cleanUp => cleanUp.length > 1);
```

Comment:
you can split the property and check if you have the property name and value on your reduce callback. There will be -2 loops